test/new-e2e/pkg/runner/profile.go

```diff
@@ -106,6 +118,57 @@ func (p baseProfile) SecretStore() parameters.Store {
 	return p.secretStore
 }
 
+// GetOutputDir returns the root output directory for tests to store output files and artifacts.
+// The directory is created on the first call to this function, normally this will be when a
+// test calls GetTestOutputDir.
+//
+// The root output directory is chosen in the following order:
+//   - outputDir parameter from the runner configuration, or E2E_OUTPUT_DIR environment variable
+//   - default provided by a parent profile, <defaultOutputRootFolder>/e2e-output, e.g. $CI_PROJECT_DIR/e2e-output
+//   - os.TempDir()/e2e-output
+//
+// A timestamp is appended to the root output directory to distinguish between multiple runs,
+// and os.MkdirTemp() is used to avoid name collisions between parallel runs.
+//
+// See GetTestOutputDir for a function that returns a subdirectory for a specific test.
+func (p baseProfile) GetOutputDir() (string, error) {
+	if p.outputRootFolder == "" {
+		var outputRoot string
+		configOutputRoot, err := p.store.GetWithDefault(parameters.OutputDir, "")
+		if err != nil {
+			return "", err
+		}
+		if configOutputRoot != "" {
+			// If outputRoot is provided in the config file, use it as the root directory
+			outputRoot = configOutputRoot
+		} else if p.defaultOutputRootFolder != "" {
+			// If a default outputRoot was provided, use it as the root directory
+			outputRoot = filepath.Join(p.defaultOutputRootFolder, "e2e-output")
+		} else if outputRoot == "" {
+			// If outputRoot is not provided, use os.TempDir() as the root directory
+			outputRoot = filepath.Join(os.TempDir(), "e2e-output")
+		}
+		// Append timestamp to distinguish between multiple runs
+		// Format: YYYY-MM-DD_HH-MM-SS
+		// Use a custom timestamp format because Windows paths can't contain ':' characters
+		// and we don't need the timezone information.
+		timePart := time.Now().Format("2006-01-02_15-04-05")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Wouldn't this increase local storage ? Have you considered tooling to clean it up ?

Reply:
Valid for non temp folder, as temp folder are cleaned up at os restart